bin/resource_management/url_database.py

```diff
@@ -45,110 +46,193 @@ def check_for_connection():
         except (requests.ConnectionError, requests.Timeout) as exception:
             print("No internet connection.")
 
-print("Starting...")
-print("Reading and testing URL's")
-
-for i, dataset in enumerate(open(file), 1):
-    line = json.loads(dataset)
-    resources = line["resources"]
-    for l in range(len(resources)):
-        url = resources[l]["url"].encode('utf-8')
-        if url in urls:
+def get_batch_response(batch_urls, i, prev_i):
+    sys.stderr.write("\r")
+    sys.stderr.write("Testing Datasets {0} - {1}".format(prev_i, i))
+    headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"}
+    rs = (grequests.head(u, timeout=60, headers=headers, verify=False, allow_redirects=True, stream=False) for u in
+          batch_urls)
+    batch_response = grequests.map(rs)
+    for j, r in enumerate(batch_response):
+        if not r is None:
+            r.close()
+        else:
+            batch_response[j] = "N/A"
+    return batch_response
+
+def get_batch_response_ftp(batch_urls):
+    ftp_responses = []
+    ftp_dates = []
+    requests_ftp.monkeypatch_session()
+    headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"}

```

Comment:
you can use a descriptive name for this value and define it at the top of the module like this:
```python
# use a generic user agent to detect services that render information pages instead of the actual data when a web browser user visits
GENERIC_WEB_CLIENT_AGENT = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
```